Datasets/LanguageModelDataset.swift

```diff
@@ -1,65 +1,57 @@
 import TensorFlow
 
-//Build a dataset suitable for language modeling from an array of texts
-public struct LanguageModelDataset<Item> {
-  public typealias Index = Int
-  public typealias Element = TensorPair<Int32, Int32>
-
-  //A function that reads Item to get an array of Int
-  public let openItem: (Item) -> [Int]
-  //The size of a batch
+/// A dataset suitable for language modeling.
+public struct LanguageModelDataset<Texts> 
+where Texts: Collection, Texts.Index==Int, Texts.Element==[Int] {
+  /// The size of a batch.
   public var batchSize: Int
-  //The length of a sequence
+  /// The length of a sequence.
   public var sequenceLength: Int
-  //The array of raw items to use
-  public let items: [Item]
-  //The length of each processed item
+  /// The array of raw items to use.
```

Comment:
I think this doc comment could use improvement.
I think this is an improvement:
```suggestion
  /// The collection of numericalized texts.
```
But still not great. What does "numericalized" means?

Reply:
numericalized is a standard NLP text. Preprocessing raw texts has two steps:
- tokenization, which is splitting a text in tokens (either characters, words, or part of words depending on the tokenizer)
- numericalization, which is converting those tokens to numbers using a vocab.